From 82d3c6f814f97a1bdfcb6008cfde91365925268a Mon Sep 17 00:00:00 2001
From: ejaaskel <esa.jaaskela@suomi24.fi>
Date: Tue, 9 Dec 2025 14:34:43 +0300
Subject: [PATCH] policy/modules/system/logging: Fixes for syslog-ng

Fix SELinux failures for syslog-ng. This patch should allow most of the
requested things, with the exception of network connection, which should
be manually enabled if remote logging is enabled, and writing to root_t,
which sounds like it should not be allowed in the first place.

Upstream-Status: Inappropriate [Sulka specific]
---
 policy/modules/system/logging.te | 4 ++++
 1 file changed, 4 insertions(+)

diff --git a/policy/modules/system/logging.te b/policy/modules/system/logging.te
index 5fa90355f..12fefffc9 100644
--- a/policy/modules/system/logging.te
+++ b/policy/modules/system/logging.te
@@ -456,6 +456,10 @@ manage_files_pattern(syslogd_t, syslogd_var_lib_t, syslogd_var_lib_t)
 allow syslogd_t syslogd_var_lib_t:file map;
 files_search_var_lib(syslogd_t)
 
+allow syslogd_t device_t:file { append create getattr open setattr };
+allow syslogd_t etc_t:dir watch;
+allow syslogd_t tmpfs_t:dir search;
+
 # manage runtime files
 allow syslogd_t syslogd_runtime_t:dir create_dir_perms;
 allow syslogd_t syslogd_runtime_t:sock_file manage_sock_file_perms;
